File: apps/api/app/models/citation.py
"""ORM model: citation — DATA_MODEL.md §5 Sources.
Written first because other tables reference it as FK.
4 seed rows — must NOT be conflated (Charaka, Yogaratnakara, Madhava, Sushruta).
"""
import uuid
from sqlalchemy import String, Text
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base


class Citation(Base):
    __tablename__ = "citation"

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    rule_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    source_text: Mapped[str] = mapped_column(String(255), nullable=False)
    chapter: Mapped[str | None] = mapped_column(String(255), nullable=True)
    verse: Mapped[str | None] = mapped_column(String(100), nullable=True)
    quote_en: Mapped[str] = mapped_column(Text, nullable=False)

File: packages/schema/src/citation.ts
// Citation — DATA_MODEL.md §Core Tables: `citation` + §5 source table
// Every derived clinical value must cite its source.
// Charaka / Yogaratnakara / Madhava Nidana / Sushruta — never conflated.

export interface Citation {
  id: string;
  /** Internal rule or computation ID that uses this citation */
  rule_id: string;
  source_text: string; // e.g. "Charaka Samhita"
  chapter?: string;    // e.g. "Vimanasthana ch.8"
  verse?: string;
  quote_en: string;
}

// Seed IDs — used as FK in other types; values populated in seed/citation_seed.py
export const CITATION_IDS = {
  CHARAKA_10PT:    "cite_charaka_vimanasthana_8",
  YOGARATNAKARA_8PT: "cite_yogaratnakara_ashtasthana",
  MADHAVA_5PT:     "cite_madhava_nidana_panchaka",
  SUSHRUTA_6STAGE: "cite_sushruta_sutrasthana_21",
} as const;
